### PR TITLE
highlight with ansi color (for console)

### PR DESCRIPTION
**Pull request by privat**
Because the nit interactive interpreter and nitx might want it

capture du 2017-11-29 14-05-07

This provides a new `AnsiHighlightVisitor` than can do stuff

Note: colors are currently hard-coded and only comments, keywords, classnames and literals are highlighted. However, since this reuse the full highlighting infrastructure, a lot of semantic things can be added.

### CODE REVIEW

File: src/highlight.nit [hunk 1]
```diff
@@ -432,6 +387,51 @@ class HighlightVisitor
...
+
+		# Prepare an stand-alone model and model builder.
+		# Unfortunately, models are enclosing and append-only.
+		# There is no way (yet?) to have a shared module `core` with
+		# isolated and throwable user modules.
+		var model = new Model
+		var mb = new ModelBuilder(model, tc)
+
+		# Parse the code
+		var source = new SourceFile.from_string("", content + "\n")
```
Comment (Morriar) [inline src/highlight.nit]: I had to do the same thing once, maybe we need a better abstraction for pieces of code that are not source files.

File: src/highlight.nit [hunk 2]
```diff
@@ -109,3 +110,48 @@ class AbstractHighlightVisitor
...
+	# The produced highlighting
+	var result = new Template
+
+	redef fun do_highlight(f, l)
+	do
+		var c
+		c = f
+		while c != null do
+			if c != f then result.add(c.blank_before)
+			result.add c.ansi_colored
```
Comment (Morriar) [inline src/highlight.nit]: Maybe use double dispatch so the Visitor holds the colors settings?

File: tests/sav/nitlight_args4.res [hunk 3]
```diff
@@ -0,0 +1,66 @@
+[34m# This file is part of NIT ( http://www.nitlanguage.org ).
```
Comment (Morriar) [inline tests/sav/nitlight_args4.res]: Should we add a `--no-color` option? :p

File: src/highlight.nit (hunk 2 above)
Comment (privat) [reply src/highlight.nit]: in a future PR :)

File: src/highlight.nit (hunk 1 above)
Comment (privat) [reply src/highlight.nit]: A source-file needs a name. https://pm1.narvii.com/6546/9158ae37079f9363f1d9f3346357c9ffe0fa3f38_hq.jpg

**Comment by privat** [comment]
test tools please

**Pull request merged**